### PR TITLE
[KYUUBI-120]define interface class IKyuubiOperation to allow different implementation of kyuubiOperation #120

### PR DESCRIPTION
**Pull request by turboFei**

### CODE REVIEW

**Comment by hustfeiwang** [comment]
KYUUBI-120 Done.

**Comment by hustfeiwang** [comment]
I have added comments for IKyuubiOperation.

File: kyuubi-server/src/main/scala/yaooqinn/kyuubi/operation/KyuubiClientOperation.scala [hunk 1]
```diff
@@ -452,43 +321,21 @@ class KyuubiOperation(session: KyuubiSession, statement: String) extends Logging
...
-      false
-    } else {
-      // check only when it's in terminal state
-      state.isTerminal && lastAccessTime + operationTimeout <= System.currentTimeMillis()
-    }
-  }
 }
 
-object KyuubiOperation {
+object KyuubiClientOperation {
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/yaooqinn/kyuubi/operation/KyuubiClientOperation.scala]: move this to object IKyuubiOperation or package object

Comment (turboFei) [reply kyuubi-server/src/main/scala/yaooqinn/kyuubi/operation/KyuubiClientOperation.scala]: done.

**Comment by yaooqinn** [comment]
please fix conflicts，we are going to merge this in v0.7

**Comment by turboFei** [comment]
> please fix conflicts，we are going to merge this in v0.7

Done.

File: kyuubi-server/src/main/scala/yaooqinn/kyuubi/operation/IKyuubiOperation.scala [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+   * Run this IKyuubiOperation.
+   * @throws KyuubiSQLException
+   */
+  @throws[KyuubiSQLException]
+  def run(): Unit
+
+  /**
+   * Check whether this IKyuubiOperation is closed or has been canceled.
+   */
+  def isClosedOrCanceled: Boolean
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/yaooqinn/kyuubi/operation/IKyuubiOperation.scala]: this one seem not a public api

Comment (turboFei) [reply kyuubi-server/src/main/scala/yaooqinn/kyuubi/operation/IKyuubiOperation.scala]: done.

File: kyuubi-server/src/main/scala/yaooqinn/kyuubi/operation/AbstractKyuubiOperation.scala [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+  protected var state: OperationState = INITIALIZED
+  protected val opHandle: OperationHandle =
+    new OperationHandle(EXECUTE_STATEMENT, session.getProtocolVersion)
+  protected val operationTimeout: Long
+  protected var lastAccessTime = System.currentTimeMillis()
+
+  protected var hasResultSet: Boolean = false
+  protected var operationException: KyuubiSQLException = _
+  protected var backgroundHandle: Future[_] = _
+  protected var statementId: String = _
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/yaooqinn/kyuubi/operation/AbstractKyuubiOperation.scala]: i guess statementId could be a val

**Pull request closed without merging**